operators.js

```diff
@@ -107,16 +107,21 @@ function equal(seek, target, opts) {
   const value = toBufferOrFalsy(target)
   const valueName = !value ? '' : value.toString()
   const indexType = opts.indexType
-  const indexName = opts.prefix
-    ? safeFilename(indexType)
-    : safeFilename(indexType + '_' + valueName)
+  const indexName = safeFilename(
+    opts.prefix
+      ? opts.useMap
+        ? indexType + '_map'
+        : indexType
```

Comment:
I think this part is where my main review is. Suppose we have a query like `slowEqual('value.content.geolocation.map', 'mercator', { prefix: 32 })`. But assuming that there are no schema enforcement in SSB, there would also be `slowEqual('value.content.geolocation', 'mercator', { prefix: 32 })`

Then prior to this PR, the `indexName` for those would be:

- For ``slowEqual('value.content.geolocation.map', 'mercator', { prefix: 32 })`
  - value_content_geolocation_map`
- For `slowEqual('value.content.geolocation', 'mercator', { prefix: 32 })`
  - `value_content_geolocation`

But after this PR, if we change

```diff
-slowEqual('value.content.geolocation', 'mercator', { prefix: 32 })
+slowEqual('value.content.geolocation', 'mercator', { prefix: 32, useMap: true })
```

Then the `indexName`s would be 

- For ``slowEqual('value.content.geolocation.map', 'mercator', { prefix: 32 })`
  - `value_content_geolocation_map`
- For `slowEqual('value.content.geolocation', 'mercator', { prefix: 32, useMap: true })`
  - `value_content_geolocation_map`

Hence a collision, potentially some bugs there. So I think we should find a different convention.

Perhaps this is enough?

```diff
-_map
+__map
```

Then we know that one underscore symbolizes one object `.` operation, and two underscores (can't do `..` operation in JS) symbolizes something else, kind of metadata.

Reply:
Good idea